### PR TITLE
[BD-46] feat: useIndexOfLastVisibleChild, responsive tabs

### PR DESCRIPTION
**Pull request by monteri**
## Description

1. Added `useIndexOfLastVisibleChild` hook:
- got code from [code](https://github.com/openedx/frontend-app-learning/blob/907892e7bb1263d6b9b56d8acc7fe57abd34cbdd/src/generic/tabs/useIndexOfLastVisibleChild.js#L33) and modified
- added support for dynamic style changes on the site
- updated parameters for hook to be able to use it when it is not possible to pass ref straightly to the container element or to the overflow element
- added tests
2. Added code for the `Tabs` component to use new hook and support responsive
- `useMemo` for `children` to update them when hook returns different index
- click on the dropdown item activates corresponding tab
- observer tracks tab change and update active state in the tab
- tab styles
- interactions with tabs using keyboard

JIRA: [PAR-805](https://openedx.atlassian.net/browse/PAR-805)

### Deploy Preview

[Preview link](https://deploy-preview-1287--paragon-openedx.netlify.app)

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [ ] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @monteri! I've created [OSPR-6686](https://openedx.atlassian.net/browse/OSPR-6686) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by monteri** [comment]
@wittjeff @adamstankiewicz 
I have opened PR and it requires your accessibility and styling review on the current state of Responsive Tabs.

**Review by adamstankiewicz — COMMENTED** [review]
Awesome work on this so far! I've added some questions/suggestions.

File: src/Tabs/index.jsx [hunk 1]
```diff
@@ -27,27 +78,89 @@ const Tabs = ({
       } else {
         newTitle = title;
       }
-      const modifiedTab = React.cloneElement(child, { ...rest, title: newTitle });
+      const tabClass = index > indexOfLastVisibleChild ? 'pgn__tab_invisible' : '';
```
[the remark this answers is not in the document]
Comment (monteri) [reply src/Tabs/index.jsx:81]: `Tab` component, unfortunately doesn't support spread operation, so additional attributes are ignored.

File: src/Tabs/index.jsx [hunk 2]
```diff
@@ -27,27 +78,89 @@ const Tabs = ({
...
+        </Dropdown.Item>
+      ));
+
+    childrenList.splice(indexOfOverflowStart, 0, (
+      <Tab
+        key="moreTabKey"
+        tabClassName={classNames(!overflowChildren.length && 'pgn__tab_invisible', 'pgn__tab_more')}
+        title={(
+          <Dropdown ref={overflowElementRef}>
+            <Dropdown.Toggle
```
[the remark this answers is not in the document]
Comment (monteri) [reply src/Tabs/index.jsx:111]: Current state: active state is added for `More tab`. Functionality works good but styles require additional effort

[the remark this answers is not in the document]
Comment (monteri) [reply src/Tabs/index.jsx:111]: Empty bubble is added for `More tab` is any notification inside the dropdown

File: src/Tabs/index.jsx [hunk 3]
```diff
@@ -1,21 +1,72 @@
...
+    overflowElementRef.current?.parentNode,
+  );
+
+  useEffect(() => {
+    if (containerElementRef.current) {
+      const observer = new MutationObserver((mutations => {
+        mutations.forEach(mutation => {
+          const eventKey = mutation.target.getAttribute('data-rb-event-key');
+          const isActive = mutation.target.getAttribute('aria-selected') === 'true';
+          const element = containerElementRef.current.querySelector(`[datakey='${eventKey}']`);
```
[the remark this answers is not in the document]
Comment (monteri) [reply src/Tabs/index.jsx:34]: Added some comments to clarify attributes

File: src/Tabs/index.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamstankiewicz) [reply src/Tabs/index.jsx:92]: Got it. Let's hold off on `aria-hidden="true"` for now then. I'll backlog an issue to check to see if it's actually necessary.

File: src/Tabs/index.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamstankiewicz) [reply src/Tabs/index.jsx:127]: Looks good enough for now, imo! We'll work with design to figure out how to improve the styles from here.

**Review by adamstankiewicz — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@monteri 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1287&entry.1484655318=https://github.com/monteri&entry.752974735=2022-05-17+08:49&entry.1917517419=2022-05-27+10:41&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**